write-fonts/src/offsets.rs

```diff
@@ -44,7 +44,7 @@ impl<T, const N: usize> std::ops::DerefMut for OffsetMarker<T, N> {
 }
 
 impl<T, const N: usize> std::ops::Deref for NullableOffsetMarker<T, N> {
-    type Target = Option<T>;
+    type Target = Option<Box<T>>;
```

Comment:
This is a bit unfortunate but I don’t see a better… option. 

I believe adding `as_ref(&self) -> Option<&T>` to this type would avoid the changes outside this file if we want to reduce the diff.

Reply:
to elaborate on this a tiny bit:


if you add the following fns to `NullableOffsetMarker`:

```rust
/// Convert from `&NullableOffsetMarker<T>` to `Option<&T>`.
    pub fn as_ref(&self) -> Option<&T> {
        match &self.obj {
            Some(obj) => Some(&*obj),
            None => None,
        }
    }

    /// Convert from `&NullableOffsetMarker<T>` to `Option<&T>`.
    pub fn as_mut(&mut self) -> Option<&mut T> {
        match &mut self.obj {
            Some(obj) => Some(&mut *obj),
            None => None,
        }
    }
```

Then the only other file you need to change is `validate.rs`, and there it's a couple of lines (basically make the body of the `Validate` impl for this type look like the one for `Option<T>`, directly below it)